File: 2015/day_01/ruby/attempt_1.rb
input = File.read("../input.txt").strip

final_floor = input.chars.sum { |c| c == '(' ? 1 : -1 }
puts "Part 1: Final floor = #{final_floor}"

floor = 0
position = nil

input.chars.each_with_index do |c, i|
  floor += (c == '(' ? 1 : -1)
  if floor == -1
    position = i + 1
    break
  end
end

puts "Part 2: First basement position = #{position}"

File: 2015/day_01/rust/attempt_1.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("../input.txt").expect("Failed to read file");

    let final_floor: i32 = input.chars().map(|c| match c {
        '(' => 1,
        ')' => -1,
        _ => 0,
    }).sum();

    println!("Part 1: Final floor = {}", final_floor);

    let mut floor = 0;
    let mut position = None;

    for (i, c) in input.chars().enumerate() {
        match c {
            '(' => floor += 1,
            ')' => floor -= 1,
            _ => {}
        }
        if floor == -1 {
            position = Some(i + 1);
            break;
        }
    }

    println!("Part 2: First basement position = {}", position.unwrap());
}
